components/authz-service/server/v2/system_integration_test.go

```diff
@@ -317,23 +226,165 @@ func TestIntegrationListRulesForAllProjects(t *testing.T) {
 			actualPid1Rules, ok := resp.ProjectRules[pid1]
 			require.True(t, ok)
 			actualPid2Rules, ok := resp.ProjectRules[pid2]
 			require.True(t, ok)
-			assert.ElementsMatch(t, expectedRules1, actualPid1Rules.Rules)
-			assert.ElementsMatch(t, expectedRules2, actualPid2Rules.Rules)
+			rulesEqual(t, expectedRules1, actualPid1Rules.Rules)
+			rulesEqual(t, expectedRules2, actualPid2Rules.Rules)
+
+			// Manually trigger completed state and wait for it to be true
+			eventData := ts.LatestEvent.Data
+			projectUpdateIDTag := eventData.Fields[project_update_tags.ProjectUpdateIDTag].GetStringValue()
+
+			infraStatusEvent := testhelpers.CreateStatusEventMsg(projectUpdateIDTag,
+				0.0,  // EstimatedTimeCompleteInSec
+				1.0,  // percentageComplete
+				true, // completed
+				event_ids.InfraClientRunsProducerID)
+			err = ts.ProjectUpdateManager.ProcessStatusEvent(infraStatusEvent)
+			assert.NoError(t, err)
+
+			eventData = ts.LatestEvent.Data
+			projectUpdateIDTag = eventData.Fields[project_update_tags.ProjectUpdateIDTag].GetStringValue()
+
+			complianceStatusEvent := testhelpers.CreateStatusEventMsg(projectUpdateIDTag,
+				0.0,  // EstimatedTimeCompleteInSec
+				1.0,  // percentageComplete
+				true, // completed
+				event_ids.ComplianceInspecReportProducerID)
+			err = ts.ProjectUpdateManager.ProcessStatusEvent(complianceStatusEvent)
+			assert.NoError(t, err)
+
+			testhelpers.WaitForWithTimeout(t, func() bool {
+				return config.NotRunningState == ts.ProjectUpdateManager.State()
+			}, time.Second*3, "State did not switch to Running")
+
+			// Update and delete rules
+			_, err = ts.Projects.DeleteRule(ctx, &api_v2.DeleteRuleReq{Id: id1})
+			require.NoError(t, err)
+
+			newConditions := []*api_v2.Condition{
+				{
+					Attribute: api_v2.ProjectRuleConditionAttributes_CHEF_TAGS,
+					Values:    []string{"test_tag1", "test_tag2"},
+					Operator:  api_v2.ProjectRuleConditionOperators_MEMBER_OF,
+				},
+				{
+					Attribute: api_v2.ProjectRuleConditionAttributes_CHEF_ORGS,
+					Values:    []string{"test_org"},
+					Operator:  api_v2.ProjectRuleConditionOperators_EQUALS,
+				},
+			}
+
+			_, err = ts.Projects.UpdateRule(ctx, &api_v2.UpdateRuleReq{
+				Id:         createResp2.Rule.Id,
+				ProjectId:  createResp2.Rule.ProjectId,
+				Name:       "this has been updated",
+				Type:       createResp2.Rule.Type,
+				Conditions: newConditions,
+			})
+			require.NoError(t, err)
+
+			// Re-apply and check cache
+			_, err = ts.Projects.ApplyRulesStart(ctx, &api_v2.ApplyRulesStartReq{})
+			assert.NoError(t, err)
+
+			expectedAfterApplyRules1 := []*api_v2.ProjectRule{
+				&api_v2.ProjectRule{
+					Id:         createResp2.Rule.Id,
+					ProjectId:  createResp2.Rule.ProjectId,
+					Name:       "this has been updated",
+					Type:       createResp2.Rule.Type,
+					Conditions: newConditions,
+					Status:     "applied",
+				},
+			}
+			eexpectedAfterApplyRules2 := []*api_v2.ProjectRule{createResp3.Rule}
```

Comment:
s/eexpected/expected

i've lost track of which of these responses should be missing the deleted rule